rskj-core/src/main/java/co/rsk/peg/federation/FederationStorageProviderImpl.java

```diff
@@ -367,6 +372,11 @@ private void savePendingFederation(ActivationConfig.ForBlock activations) {
         bridgeStorageAccessor.saveToRepository(PENDING_FEDERATION_KEY.getKey(), serializedPendingFederation);
     }
 
+    private void saveProposedFederation(Federation proposedFederation) {
+        saveFederationFormatVersion(PROPOSED_FEDERATION_FORMAT_VERSION.getKey(), proposedFederation.getFormatVersion());
```

Comment:
since the proposed federation is something added after rskip123, there's no need to check that when saving the format version

Reply:
What if the `setProposedFederation(Federation)` method is called with `null`? This line would fail.

Should we reset the `isProposedFederationSet` flag if the `setProposedFederation(Federation)` method is called with `null`?

Reply:
good catch jere. Thanks. This requires some analysis.

Reply:
solved.